Mage/src/main/java/mage/abilities/costs/common/PayLoyaltyCost.java

```diff
@@ -33,8 +34,19 @@ public PayLoyaltyCost(PayLoyaltyCost cost) {
 
     @Override
     public boolean canPay(Ability ability, Ability source, UUID controllerId, Game game) {
+        int loyaltyCost = amount;
+        if (ability instanceof LoyaltyAbility) {
+            LoyaltyAbility copiedAbility = ((LoyaltyAbility) ability).copy();
+            game.getContinuousEffects().costModification(copiedAbility, game);
+            loyaltyCost = 0;
+            for (Cost cost : copiedAbility.getCosts()) {
+                if (cost instanceof PayLoyaltyCost) {
+                    loyaltyCost += ((PayLoyaltyCost) cost).amount;
```

Comment:
All loyalty cost modification must be moved to standard method with normal cost modification:
* `public void costModification(Ability abilityToModify, Game game)`

Or maybe there were a reason for your workaround (for special loyalty modification code)?

Reply:
Not sure what you mean here.  The issue is that the costModification method wasn't getting called before canPay does so I called it in line 40.

All this code you quoted does is get the new cost (after modification) and uses it to check if the planeswalker has enough loyalty counters.